43 create volunteer page hero

ogorman89 opened the pull request:
## This PR:

resolves #43 and resolves #42 by creating a reusable `Hero` component for use on any page.

- I created a reusable `Hero` component
- - This allows `heroText`, `heroImage`, and `pageName` to be passed to `Hero` component via `props`
- Moved `Hero` to `/components/global` & updated imports where necessary
- Added a folder `/Hero` for the component and styling to live since it clutters up the component a ton
- Moved `Hero` styles to a separate `styles.js` in the `/Hero` folder to make it more readable

I figured a reusable component would be best in case we decide to add new pages to the site later on. 🤷‍♂️

---

## Screenshots (if applicable):

https://github.com/codeforpdx/codepdx_website/assets/3335277/1a9a5663-992b-46c2-8807-e3a3a038532e

---

## Future Steps/PRs Needed to Finish This Work (optional):

The component styling needs some work.  I'm not happy with the styling especially at a few weird breakpoints, but I'd like to talk with the group before spending a lot of time fussing with the mobile layouts/scaling, etc. Perhaps we can add it to the agenda for the next in-person meeting?

ogorman89 committed:
add projects and volunteers images, add props to main pages, use props to conditionally render hero section

ogorman89 committed:
move images and text to props

ogorman89 committed:
move Hero to global components, move hero styles to seperate file, add styles for each page variation, update imports

xscottxbrownx commented on src/components/global/Hero/Hero.jsx:
```diff
@@ -0,0 +1,47 @@
+//import component specific styles
+import { heroStyles } from './styles';
+
+const Hero = (props) => {
```
Personally, I would prefer destructuring these `props` to `{ pageName, heroImage, heroText }` that way any dev knows exactly what this component expects - without having to dig into it.  

Unless this way is easier for you to tell what is imported through the styles file vs passed down as a prop, then leave as is - just an opinion.

---

Either way, eslint will continue to whine about it not knowing what type these props are.  Do you happen to be familiar with PropTypes? Declaring the types using PropTypes would solve, and give other devs another clue to not only what props this component expects - but what those props should be.

ogorman89 replied:
Wasn't familiar with PropTypes but I imported it (already a dependency) and added propTypes to the bottom of Hero right before the export. Please let me know if it looks correct. It seems to work 🤷‍♂️

xscottxbrownx commented on src/components/global/Hero/Hero.jsx:
```diff
@@ -0,0 +1,47 @@
+          {props.heroText}
+        </Typography>
+        {props.pageName === 'home' && (
+          <Button
```
Since this is a `mailto`, it doesn't need the `target` or `rel` to open in separate window - it opens in whatever email client user has.  I don't BELIEVE they are needed for anything security-wise either, but could be wrong.

Jared-Krajewski replied:
To my knowledge you are correct. they are not needed for `mailto`. I do want to eventually use a web form for the email to prevent phishing or webscraping spam to Hugh's email. Also a lot of people don't have a default mail app set up on their computer so it's not the nicest solution, just the easiest for now. we can talk more about this at meetup.

ogorman89 replied:
I removed the target and rel in the last push.

xscottxbrownx commented on src/pages/Projects.jsx:
```diff
@@ -8,7 +8,11 @@ const Projects = () => {
     // or use <Box component="img" />
     <>
-      <Hero />
+      <Hero
```
I find this photo to be pretty dark - testing on all of them I think we should change ALL the gradient opacities to be `0.3` vs the `0.5` we have.

Jared-Krajewski replied:
should still have plenty of contrast. If not we can figure it out at the meeting.

ogorman89 replied:
I changed to 0.3 for now I think it looks good!

xscottxbrownx commented on src/pages/Projects.jsx:
```diff
@@ -8,7 +8,11 @@ const Projects = () => {
     // either make Hero reusable somehow (and move it to components/global)
```
Can delete all the comments, now that you figured out direction.

ogorman89 replied:
removed.

xscottxbrownx commented on src/pages/Volunteer.jsx:
```diff
@@ -8,8 +8,11 @@ const Volunteer = () => {
     // either make Hero reusable somehow (and move it to components/global)
```
Can delete all the comments, now that you figured out direction.

ogorman89 replied:
removed.

xscottxbrownx commented on src/components/global/Hero/styles.js:
```diff
@@ -0,0 +1,69 @@
+        width: { xs: '80%', md: '100%' },
+        textAlign: 'center'
+    },
+    projectsH1: {
```
Find this one little tough to read on small screens, we can discuss at meeting.

Jared-Krajewski replied:
Text should be white to match the rest. It just wasn't updated in Figma. It will also need top padding lowered to stop it from pushing out of the hero on small screens. 

<img width="656" alt="Screenshot 2023-11-05 at 2 32 16 PM" src="https://github.com/codeforpdx/codepdx_website/assets/114430929/758605f0-a195-469a-a41d-9d3596f32c21">

![Screenshot 2023-11-05 at 2 35 09 PM](https://github.com/codeforpdx/codepdx_website/assets/114430929/b6714f67-36ea-47a6-9270-66c5c8009876)

ogorman89 replied:
text color updated!

xscottxbrownx reviewed (commented):
All in all, looking good! 

Let me know if you have questions / what you think of comments provided.

Jared-Krajewski reviewed (changes requested):
Overall looks good. In agreement with Scott's comments as well.

ogorman89 committed:
destructure Hero props

ogorman89 committed:
add prop-types to Hero

ogorman89 committed:
remove target and rel from mailto link

ogorman89 committed:
update opacity on hero images from 0.5 to 0.3

ogorman89 committed:
remove TODO comments

ogorman89 committed:
make projects h1 white

ogorman89 commented:
I've addressed all the comments. First time adding PropTypes so hopefully, that looks right to you all.

Jared-Krajewski reviewed (approved):
Looks good to me. Anything further on design we can talk about at meeting.

andycwilliams commented on src/components/global/Hero/styles.js:
```diff
@@ -0,0 +1,71 @@
+export const heroStyles = {
```
Could we rename the file to `heroStyles.js`, just for a bit more clarity? In fact, I wonder if we can simply move all this to `theme.js`.

andycwilliams commented on src/pages/Projects.jsx:
```diff
@@ -1,15 +1,16 @@
+      <Hero
+        pageName={'projects'}
+        heroImage={`linear-gradient(rgba(0, 0, 0, 0.3), rgba(0, 0, 0, 0.3)), url(/assets/projectsHeroImage.png)`}
+        heroText={`Our products blend innovation, quality, and user-centric design to meet today's needs and anticipate tomorrow's challenges`}
```
Really nitpicky, but maybe we could center the text so that they all begin on the exact same level? For example, the volunteer page text appears slightly above where the projects page text does. Not important, though.

andycwilliams commented on src/components/global/Hero/styles.js:
```diff
@@ -0,0 +1,71 @@
+        textTransform: 'capitalize',
+        fontWeight: '200',
+        color: 'black',
+        textShadow: '0px 4px 4px #0000004D',
```
Can we remove this shadow? Just this one, not the `h1`s. This is what's in the Figma but I'm not super fond of it. Other buttons don't have this styling and it makes the text slightly tougher to distinguish, in my opinion.

andycwilliams commented on src/components/global/Hero/styles.js:
```diff
@@ -0,0 +1,71 @@
+    },
+    contactBtnStyle: {
+        textTransform: 'capitalize',
+        fontWeight: '200',
```
Is `fontWeight` really carrying its (font) weight here? I'm not seeing a significant difference with it. I have to bump it up to 600 or so for a noticeable change.

andycwilliams commented on src/components/global/Hero/styles.js:
```diff
@@ -0,0 +1,71 @@
+        textShadow: '0px 4px 4px #0000004D',
+        alignSelf: 'center',
+        mt: '2em',
+        fontSize: { xs: '15px', md: '18px' },
```
I'd like the font size to be a tad larger, so it fills out the button space a little more.

andycwilliams reviewed (approved):
I have no problem approving this. My nitpicks and suggestions are more to get my thoughts down for discussion during the meeting, not things I demand.

xscottxbrownx commented:
> I've addressed all the comments. First time adding PropTypes so hopefully, that looks right to you all.

Almost...

Anything the component absolutely must be passed as props for component to work needs to be marked as required. So, I THINK currently the way you have this component coded - it needs all 3 of the specific props.  COULD code it so the text is conditionally rendered if heroText is passed as a prop - so that one wouldn't need to be required but up to you.

marking as required is just adding on `.isRequired` at the end so:
`heroImage: PropTypes.string.isRequired`,

ogorman89 commented:
> > I've addressed all the comments. First time adding PropTypes so hopefully, that looks right to you all.
> 
> Almost...
> 
> Anything the component absolutely must be passed as props for component to work needs to be marked as required. So, I THINK currently the way you have this component coded - it needs all 3 of the specific props. COULD code it so the text is conditionally rendered if heroText is passed as a prop - so that one wouldn't need to be required but up to you.
> 
> marking as required is just adding on `.isRequired` at the end so: `heroImage: PropTypes.string.isRequired`,

That makes sense thanks for explaining. I made the text optional and changed the Typography to conditionally render if the prop is included.

ogorman89 commented:
@andycwilliams, thanks for reviewing. Your design comments are spot on. I will wait to implement styling changes until our in-person meeting. I hope we can do a full review and write up some styling issues because this is not my strong suit. I am sure I have CSS rules that need to be revised in my current version. 

As for renaming the styles.js in the Hero folder, we can follow whatever naming convention the group wants. From the example code I've seen, most of this sx-level styling is kept with the component or in the same folder as the component and not in the master theme. That's just my observation of a tiny sample size. 

We can decide at the in-person meeting how we want to style components and where we want those styles to live.

ogorman89 committed:
add isRequired for required Hero props, add conditional render to heroText  of Hero component

ogorman89 committed:
Merge branch 'main' into 43-create-volunteer-page-hero
